Draft project governance documentation (#222)
carpentries/lesson-development-training: CONTRIBUTING.md

   to [The Workbench documentation][template-doc].

### What to Contribute (General)

There are many ways to contribute, from writing new exercises and improving
existing ones to updating or filling in the documentation and submitting [bug
reports][issues] about things that do not work, are not clear, or are missing.
If you are looking for ideas, please see [the list of issues for this
repository][repo], or the issues for [Data Carpentry][dc-issues], [Library
Carpentry][lc-issues], and [Software Carpentry][swc-issues] projects.

Comments on issues and reviews of pull requests are just as welcome: we are
smarter together than we are on our own. **Reviews from novices and newcomers
are particularly valuable**: it's easy for people who have been using these
lessons for a while to forget how impenetrable some of this material can be, so
fresh eyes are always welcome.

### What to Contribute (This Lesson)

The curriculum is relatively new and under active development/refinement,
so any contribution will be welcome and probably very helpful.
Ideas for how the existing content could be improved or updated,
and/or errors that need to be corrected will be particularly useful,

> but we are certainly open to suggestions for new content and all other kinds of feedback.

Reviewer: ```suggestion
and we are certainly open to suggestions for new content and all other kinds of feedback.
```

The curriculum currently lacks figures and other visual aids to teaching,
so these will be particularly appreciated.
Comments on existing issues and reviews of pull requests are similarly welcome,
whether you are a certified Lesson Developer Trainer for the program or not.

If you plan to submit a pull request, please open an issue
(or comment on an existing thread) first to ensure that effort is not duplicated
or spent making a change that will not be accepted by the Maintainers.

### What *Not* to Contribute (General)

Our lessons already contain more material than we can cover in a typical

> workshop, so we are usually *not* looking for more concepts or tools to add to

Reviewer 2: ```suggestion
workshop, so we are usually *not* looking for more content to add to
```

Author: I am going to leave this in, as it is part of the "general" section and relates to all Carpentries lessons. You could argue that "adding tools" to a lesson is a subcategory of "adding content" but I think the redundancy is worthwhile to help make the point e.g. to someone who wants to suggest adding another command to a Shell lesson or similar.

   (teaching + exercises).
3. The [learning objective(s)][cldt-lo] of this new content.
4. (optional, but strongly preferred)
   A suggestion of which of the currently-used learning objectives could be
   removed from the curriculum to make space for the new content.

### Using GitHub

If you choose to contribute via GitHub, you may want to look at [How to
Contribute to an Open Source Project on GitHub][how-contribute]. In brief, we
use [GitHub flow][github-flow] to manage changes:

1. Create a new branch in your desktop copy of this repository for each
   significant change.
2. Commit the change in that branch.
3. Push that branch to your fork of this repository on GitHub.
4. Submit a pull request from that branch to the [upstream repository][repo].
5. If you receive feedback, make changes on your desktop and push to your
   branch on GitHub: the pull request will update automatically.

NB: The published copy of the lesson is usually in the `main` branch.

Each lesson has a team of maintainers who review issues and pull requests or
encourage others to do so. The maintainers are community volunteers, and have

> final say over what gets merged into the lesson.

Reviewer 2: ```suggestion
the final say over what gets merged into the lesson.
```

Reviewer 2: Is it "the final"? Not sure.

Reviewer: think it could go either way

### Other Resources

The Carpentries is a global organisation with volunteers and learners all over
the world. We share values of inclusivity and a passion for sharing knowledge,
teaching and learning. There are several ways to connect with The Carpentries
community listed at <https://carpentries.org/connect/> including via social
media, slack, newsletters, and email lists. You can also [reach us by
email][contact].

[repo]: https://github.com/carpentries/lesson-development-training
[cldt-lo]: https://carpentries.github.io/lesson-development-training/05-objectives.html#learning-objectives

> [contact]: mailto:team@carpentries.org

Reviewer: should we update this to be a topic box specific to the CLDT Trainers?
